commands/describe.go

```diff
@@ -138,45 +133,158 @@ func getFunctionURLs(gateway string, functionName string, functionNamespace stri
 	return url, asyncURL
 }
 
-func printFunctionDescription(funcDesc schema.FunctionDescription) {
-	w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', tabwriter.TabIndent)
-	fmt.Fprintln(w, "Name:\t "+funcDesc.Name)
-	fmt.Fprintln(w, "Status:\t "+funcDesc.Status)
-	fmt.Fprintln(w, "Replicas:\t "+strconv.Itoa(funcDesc.Replicas))
-	fmt.Fprintln(w, "Available replicas:\t "+strconv.Itoa(funcDesc.AvailableReplicas))
-	fmt.Fprintln(w, "Invocations:\t "+strconv.Itoa(funcDesc.InvocationCount))
-	fmt.Fprintln(w, "Image:\t "+funcDesc.Image)
-	fmt.Fprintln(w, "Function process:\t "+funcDesc.EnvProcess)
-	fmt.Fprintln(w, "URL:\t "+funcDesc.URL)
-	fmt.Fprintln(w, "Async URL:\t "+funcDesc.AsyncURL)
-	printMap(w, "Labels", *funcDesc.Labels)
-	printMap(w, "Annotations", *funcDesc.Annotations)
-	if funcDesc.Usage != nil {
-		fmt.Println()
-
-		fmt.Fprintf(w, "RAM:\t %.2f MB\n", (funcDesc.Usage.TotalMemoryBytes / 1024 / 1024))
-		cpu := funcDesc.Usage.CPU
-		if cpu < 0 {
-			cpu = 1
+func printFunctionDescription(dst io.Writer, funcDesc schema.FunctionDescription, verbose bool) {
+	w := tabwriter.NewWriter(dst, 0, 0, 1, ' ', tabwriter.TabIndent)
+	defer w.Flush()
+
+	out := printer{
+		w:       w,
+		verbose: verbose,
+	}
+
+	process := "<default>"
+	if funcDesc.EnvProcess != "" {
+		process = funcDesc.EnvProcess
+	}
+
+	out.Printf("Name:\t%s\n", funcDesc.Name)
+	out.Printf("Status:\t%s\n", funcDesc.Status)
+	out.Printf("Replicas:\t%s\n", strconv.Itoa(int(funcDesc.Replicas)))
+	out.Printf("Available Replicas:\t%s\n", strconv.Itoa(int(funcDesc.AvailableReplicas)))
+	out.Printf("Invocations:\t%s\n", strconv.Itoa(int(funcDesc.InvocationCount)))
+	out.Printf("Image:\t%s\n", funcDesc.Image)
+	out.Printf("Function Process:\t%s\n", process)
+	out.Printf("URL:\t%s\n", funcDesc.URL)
+	out.Printf("Async URL:\t%s\n", funcDesc.AsyncURL)
+	out.Printf("Labels", *funcDesc.Labels)
+	out.Printf("Annotations", *funcDesc.Annotations)
+	out.Printf("Constraints", funcDesc.Constraints)
+	out.Printf("Environment", funcDesc.EnvVars)
+	out.Printf("Secrets", funcDesc.Secrets)
+	out.Printf("Requests", funcDesc.Requests)
+	out.Printf("Limits", funcDesc.Limits)
+	out.Printf("", funcDesc.Usage)
+}
+
+type printer struct {
+	verbose bool
+	w       io.Writer
+}
+
+func (p *printer) Printf(format string, a interface{}) {
+	switch v := a.(type) {
+	case map[string]string:
+		printMap(p.w, format, v, p.verbose)
+	case []string:
+		printList(p.w, format, v, p.verbose)
+	case *types.FunctionResources:
+		printResources(p.w, format, v, p.verbose)
+	case *types.FunctionUsage:
+		printUsage(p.w, v, p.verbose)
+	default:
+		if !p.verbose && isEmpty(a) {
+			return
 		}
-		fmt.Fprintf(w, "CPU:\t %.0f Mi\n", (cpu))
 
+		if p.verbose && isEmpty(a) {
+			a = "<none>"
+		}
+
+		fmt.Fprintf(p.w, format, a)
+	}
+
+}
+
+func printUsage(w io.Writer, usage *types.FunctionUsage, verbose bool) {
+	if !verbose && usage == nil {
+		return
+	}
+
+	if usage == nil {
```

Comment:
I would probably not print this message, since for anyone who's not on openfaas pro, they will see this all the time, so we could just omit printing it.